application/src/main/java/org/openhds/mobile/provider/DatabaseAdapter.java

```diff
@@ -255,6 +281,91 @@ public void updateCompleteStatus(long id, boolean completed) {
 		updateFormSubmission(id, cv);
 	}
 
+	//add recent forms paths
+	public long addHierarchyPath(String hierarchyPath, String filePath) {
+		long id = -1;
+		SQLiteDatabase db = dbHelper.getWritableDatabase();
+		db.beginTransaction();
+		try {
+			ContentValues cv = new ContentValues();
+			cv.put(KEY_HIERARCHY_PATH, hierarchyPath);
+			cv.put(KEY_RECENT_FORM_PATH, filePath);
+			id = db.insertOrThrow(RECENT_FORM_TABLE, null, cv);
+			db.setTransactionSuccessful();
+		} finally {
+			db.endTransaction();
+		}
+		db.close();
+		return id;
+	}
+
+	//Finds recentForms by current hierarchy
+	public Collection findFormByPath(String hierarchyPath) {
```

Comment:
Name of method changed to Collection <String> findAssociatedPath() for more clarity.